app/code/Magento/CatalogCustomerGraphQl/Model/Resolver/PriceTiers.php

```diff
@@ -120,42 +120,89 @@ function () use ($productId, $context) {
 
                 $productPrice = $this->tiers->getProductRegularPrice($productId) ?? 0.0;
                 $tierPrices = $this->tiers->getProductTierPrices($productId) ?? [];
-
-                return $this->formatProductTierPrices($tierPrices, $productPrice, $store);
+                return $this->formatAndFilterTierPrices($tierPrices, $productPrice, $store);
             }
         );
     }
 
     /**
-     * Format tier prices for output
+     * Format and filter tier prices for output
      *
      * @param ProductTierPriceInterface[] $tierPrices
      * @param float $productPrice
      * @param StoreInterface $store
      * @return array
      */
-    private function formatProductTierPrices(array $tierPrices, float $productPrice, StoreInterface $store): array
-    {
+    private function formatAndFilterTierPrices(
+        array $tierPrices,
+        float $productPrice,
+        StoreInterface $store
+    ): array {
         $tiers = [];
+        $qtyCache = [];
+
+        foreach ($tierPrices as $key => $tierPrice) {
+            $this->formatTierPrices($productPrice, $store, $tierPrice, $tiers);
+            $this->filterTierPrices($tierPrices, $key, $tierPrice, $qtyCache, $tiers);
+        }
+        return $tiers;
+    }
+
+    /**
+     * Format tier prices for output
+     *
+     * @param float $productPrice
+     * @param StoreInterface $store
+     * @param ProductTierPriceInterface $tierPrice
+     * @param array $tiers
+     */
+    private function formatTierPrices(float $productPrice, StoreInterface $store, &$tierPrice, &$tiers)
```

Comment:
we kind of prohibit passing params by reference &$tierPrice, &$tiers
You better want a function that returns something. You rarely want a function that returns 2 things.
formatTierPrices can return $tiers for example. 
Objects are passed by reference by default so you don't need  &$tierPrice to do $tierPrice->setValue
You can also do that outside formatTierPrices and don't set anything in tierPrice, just get
Plus you only need getValue and getQty  and getPercentageValue from it. You can use those params directly not by passing the whole tierPrice. 
From store you only need CurrencyCode not store.
Suddenly your method becomes: formatTierPricesValues(productPrice, tierPriceValue, TierPricePercentage, tierPriceQty): array (where array is $tiers)

Reply:
**Done**